utils/emmylua/wesnoth.lua

```diff
@@ -0,0 +1,272 @@
+---@overload fun(message:string, in_chat:boolean)
+function wesnoth.log(logger, message, in_chat) end
+
+---Simulate combat between two units_MP
+---@param attacker unit
+---@param attacker_weapon integer
+---@param defender unit
+---@param defender_weapon integer
+---@return stats_evaluation attacker_stats
+---@return stats_evaluation defender_stats
+---@return weapon_evaluation attacker_weapon
+---@return weapon_evaluation defender_weapon
+function wesnoth.simulate_combat(attacker, attacker_weapon, defender, defender_weapon) end
+
+---@class stats_evaluation
+---@field poisoned number The chance to be poisoned as a number in [0,1].
+---@field slowed number The chance to be slowed as a number in [0,1].
+---@field untouched number The chance to receive no damage as a number in [0,1].
+---@field average_hp number
+---@field hp_chance table<number, number> The chance to end the attack with a specific number of hit points. The key is the number of hit points, from 0 to max hitpoints. The value is the chance as a number in [0,1].
+
+---@class weapon_evaluation
+---@field num_blows integer The number of times the attack hits.
+---@field damage integer The damage the attack inflicts per hit.
+---@field chance_to_hit number The chance for the attack to hit.
+---@field poisons boolean Indicates whether the attack inflicts poison.
+---@field slows boolean Indicates whether the attack inflicts slow.
+---@field petrifies boolean Indicates whether the attack inflicts petrification.
+---@field plagues boolean Indicates whether the attack inflicts plague.
+---@field plague_type string If the attack inflicts plague, this specifies the unit type it is converted to.
+---@field backstabs boolean
+---@field rounds integer The number of rounds of combat.
+---@field firststrike boolean Indicates whether the attack always strikes first.
+---@field drains boolean Indicates whether the attack drains hit points.
+---@field drain_constant integer
+---@field drain_percent number
+---@field attack_num integer
+---@field name string
+
+---Construct a function that generates random names using a context-free grammar
+---@param type '"cfg"' The type of generator to create
+---@param definition table<string, string|string[]> Definition rules for the generator
+---@return fun():string
+function wesnoth.name_generator(type, definition) end
+
+---Construct a function that generates random names using Markov chains
+---@param type '"markov"' The type of generator to create
+---@param definition string[] Sample names to be input to the generator
+---@param chain_size? integer The size of the Markov chain unit
+---@param max_length? integer Maximum length of the generated names
+---@return fun():string
+function wesnoth.name_generator(type, definition, chain_size, max_length) end
+
+---Compile a WFL formula into a Lua function
+---@param formula string A WFL formula
+---@return fun(variables:WML):any
+function wesnoth.compile_formula(formula, variables) end
```

Comment:
I think this only takes 1 argument

Reply:
I might've copy-pasted `eval_formula` and edited incorrectly.